### ISSUE TITLE
ModuleNotFoundError: No module named 'bl_math'

### ISSUE THREAD

**Issue by TedThompson** [rileyzzz/BlenderIMExporter#8]
Blender v2.80 - I can't figure out how to get this resolved.

**Comment by TedThompson** [comment]
My hack was to comment out 'import bl_math' and define my own lerp, which I found on the net. 

Change all bl_math.lerp to just lerp.

Then I had to change the part where the script attempts to access .emission_color, as that's not a parameter of the principled node (at least in blender 2.8). 

I simply substituted 0,0,0 as values since a random sampling of .im files showed that's what it seems to be usually (always?) set to. 

Again, this is in Blender 2.8, perhaps these things aren't an issue in a later version.

**Comment by pwillard** [comment]
FYI: I have versions of Blender installed from 3.31 going back to V2.93 and they all can import `bl_math` without complaints.  Is there a reason you are not using a more recent version of Blender?   I can check with the specific version 2.80 quickly, as I have the kits handy. `a few moments later` YUP,  it is a specific issue with 2.80.  There is really no reason I know of that should have you stay with that version... you should at least move to one of the LTS releases of Blender.

**Comment by TedThompson** [comment]
I would humbly suggest revising the read me about version compatibility. 

Why 2.8? No reason really, I have it and I know it. My experience of being totally lost updating TO 2.8 probably fuels my lack of interest in any other updates up until now. 

This is the first thing I've hit that has given me a good reason to update. So I guess I will. 

Thank you for checking tho!

**Comment by pwillard** [comment]
For reference... from 2.8 onward, the changes are so minimal that you won't get lost.  2.79 -->2.8 in reality was like jumping from version 2 to version 16!

**Comment by TedThompson** [comment]
Which was bad for me, as I was _barely_ getting a grip on 2.79 having come from GMax, but the world moves on. 

Installed 2.93 and your .im exporter worked right out of the box, and you're right - I'm not noticing a difference in Blender.  

So, all good, back to making crappy trainz assets! :)

**Comment by rileyzzz** [comment]
Thanks for the issue report, sorry that I didn't see it until now.
Assuming the exporter all works fine in 2.93, I've incremented the addon's minimum supported version to there to prevent further confusion.
Unfortunately Blender likes to change their python API quite frequently, so it's always a bit tricky figuring out which features were added when. (Especially when I'm always working in the latest version, as I'm one of the unfortunate people who have Blender installed through Steam).
For small exporter projects like this, the modules I use are mostly stable and have been around for a while, so thankfully these issues don't pop up _too_ often, and I usually don't bother checking older blender versions. Seems like bl_math was the exception haha
